### PR TITLE
Add Blog Post for mmBERT

### PR DESCRIPTION
**Pull request by orionw**
Congratulations! You've made it this far! Once merged, the article will appear at https://huggingface.co/blog. Official articles 
require additional reviews. Alternatively, you can write a community article following the process [here](https://huggingface.co/blog-explorers).

## Preparing the Article

You're not quite done yet, though. Please make sure to follow this process (as documented [here](https://github.com/huggingface/blog/tree/main?tab=readme-ov-file#how-to-write-an-article-)):

- [x] Add an entry to [_blog.yml](https://github.com/huggingface/blog/blob/main/_blog.yml).
- [x] Add a thumbnail. There are no requirements here, but there is a [template](https://github.com/huggingface/blog?tab=readme-ov-file#how-to-get-a-nice-responsive-thumbnail) if it's helpful.
- [x] Check you use a short title and blog path.
- [x] Upload any additional assets (such as images) to the Documentation Images [repo](https://huggingface.co/datasets/huggingface/documentation-images/tree/main/blog). This is to reduce bloat in the GitHub base repo when cloning and pulling. Try to have small images to avoid a slow or expensive user experience.
- [x] Add metadata (such as authors) to your `md` file. You can also specify `guest` or `org` for the authors.
- [x] Ensure the publication date is correct.
- [x] Preview the content. A quick way is to paste the markdown content in https://huggingface.co/new-blog. **Do not click publish**, this is just a way to do an early check.

Here is an example of a complete PR: https://github.com/huggingface/blog/pull/2382

## Getting a Review

Please make sure to get a review from someone on your team or a co-author.
Once this is done and once all the steps above are completed, you should be able to merge.
There is no need for additional reviews if you and your co-authors are happy and meet all of the above.

Feel free to add @pcuenca as a reviewer if you want a final check. Keep in mind he'll be biased toward light reviews
(e.g., check for proper metadata) rather than content reviews unless explicitly asked.

### CODE REVIEW

File: mmbert.md [hunk 1]
```diff
@@ -0,0 +1,656 @@
...
+    predictions = outputs.logits[mask_indices]
+    
+    # Get top 5 predictions
+    top_tokens = torch.topk(predictions, 5, dim=-1)
+    return [tokenizer.decode(token) for token in top_tokens.indices[0]]
+
+# Example
+masked_text = "The capital of France is [MASK]."
+predictions = predict_masked_token(masked_text)
+print(f"Predictions: {predictions}")
```
Comment (tomaarsen) [inline mmbert.md]: Doesn't run, will need to be patched.

Comment (tomaarsen) [reply mmbert.md]:
```suggestion
from transformers import AutoTokenizer, AutoModelForMaskedLM
import torch

tokenizer = AutoTokenizer.from_pretrained("jhu-clsp/mmBERT-base")
model = AutoModelForMaskedLM.from_pretrained("jhu-clsp/mmBERT-base")

def predict_masked_token(text):
    inputs = tokenizer(text, return_tensors="pt")
    with torch.no_grad():
        outputs = model(**inputs)

    mask_indices = torch.where(inputs["input_ids"] == tokenizer.mask_token_id)
    predictions = outputs.logits[mask_indices]
    top_tokens, top_indices = torch.topk(predictions, 5, dim=-1)

    return [tokenizer.decode(token) for token in top_indices[0]]

# Works across languages
texts = [
    "The capital of France is <mask>.",
    "La capital de España es <mask>.",
    "Die Hauptstadt von Deutschland ist <mask>.",
]

for text in texts:
    predictions = predict_masked_token(text)
    print(f"Text: {text}")
    print(f"Predictions: {predictions}\n")
```

This should work once the models have had their tokenizers updated to:
1. set mask_token to `"<mask>"`
2. set `"lstrip": true` for `"<mask>"`, so that `"La capital de España es <mask>."` is parsed as `"La capital de España es<mask>."`

Possible with:
```python
from transformers import AutoTokenizer, AutoModelForMaskedLM, AddedToken

added_token = AddedToken(
    tokenizer.mask_token,
    lstrip=True,
    special=True,
)
tokenizer.add_special_tokens({"mask_token": added_token})
tokenizer.save_pretrained("tmp")
```

Comment (orionw) [reply mmbert.md]: Ah, needs `<mask>` not `[MASK]`

Comment (orionw) [reply mmbert.md]: Thanks so much for figuring that out @tomaarsen!!!
